crates/e2e/tests/e2e/colocation_univ2.rs

```diff
@@ -0,0 +1,270 @@
+        &web3,
+        &contracts,
+        WethPoolConfig {
+            token_amount: to_wei(1000),
+            weth_amount: to_wei(1000),
+        },
+    )
+    .await;
+    let token = token.contract;
+    tx!(
+        trader,
+        contracts.weth.approve(contracts.allowance, to_wei(3))
+    );
+    tx_value!(trader, to_wei(3), contracts.weth.deposit());
+
+    tracing::info!("Starting services.");
+    let (solver_endpoint, _solver_task) = start_solver(contracts.weth.address()).await;
+    let _driver_task = start_driver(
+        &contracts,
+        &solver_endpoint,
+        &solver.address(),
+        &SOLVER_PRIVATE_KEY,
+    );
+    let driver_url: Url = "http://localhost:11088/test_solver".parse().unwrap();
+    let _autopilot_task = start_autopilot(&contracts, driver_url);
+    let _orderbook_task = start_orderbook(&contracts);
+
+    let http = reqwest::Client::builder()
+        .timeout(Duration::from_secs(10))
+        .build()
+        .unwrap();
+
+    tracing::info!("Waiting for API to come up.");
+    let is_up = || async {
+        http.get(format!("{API_HOST}/api/v1/version"))
+            .send()
+            .await
+            .is_ok()
+    };
+    wait_for_condition(Duration::from_secs(10), is_up)
+        .await
+        .unwrap();
+
+    tracing::info!("Placing order");
+    let balance = token.balance_of(trader.address()).call().await.unwrap();
+    assert_eq!(balance, 0.into());
+    let order_a = OrderBuilder::default()
+        .with_sell_token(contracts.weth.address())
+        .with_sell_amount(to_wei(2))
+        .with_fee_amount(to_wei(1))
+        .with_buy_token(token.address())
+        .with_buy_amount(to_wei(1))
+        .with_valid_to(model::time::now_in_epoch_seconds() + 300)
+        .with_kind(OrderKind::Buy)
+        .sign_with(
+            EcdsaSigningScheme::Eip712,
+            &contracts.domain_separator,
+            SecretKeyRef::from(&SecretKey::from_slice(&TRADER_A_PK).unwrap()),
+        )
+        .build()
+        .into_order_creation();
+    let placement = http
+        .post(&format!("{API_HOST}/api/v1/orders"))
+        .json(&order_a)
+        .send()
+        .await
+        .unwrap();
+    assert_eq!(placement.status(), 201,);
+
+    tracing::info!("Waiting for trade.");
+    let trade_happened =
+        || async { token.balance_of(trader.address()).call().await.unwrap() != 0.into() };
+    wait_for_condition(Duration::from_secs(10), trade_happened)
+        .await
+        .unwrap();
+
+    let balance = token.balance_of(trader.address()).call().await.unwrap();
+    assert_eq!(balance, to_wei(1));
+
+    // TODO: test that we have other important per-auction data that should have
+    // made its way into the DB.
+}
+
+async fn start_solver(weth: H160) -> (Url, JoinHandle<()>) {
+    let config = format!(
+        r#"
+weth = "{weth:?}"
+base-tokens = []
+max-hops = 0
+        "#,
+    );
+    let mut file = tempfile::NamedTempFile::new().unwrap();
```

Comment:
We had a discussion about this in https://github.com/cowprotocol/services/pull/1198 . It is impossible to consistently delete temporary files. This is one reason I preferred a pure approach that doesn't touch disk at all.

However, usually even when the test panics, the file will be deleted because the Drop impl of NamedFile deletes it. If Drop doesn't run then usually the file will still be deleted by the operating system, which might only happen on next reboot. The files are small so this isn't an issue in practice.

Reply:
I've decided to keep panic=abort for now because it is useful to detect a background task in one of the components panicking. I might reconsider this in the future.